package.json

```diff
@@ -975,30 +975,28 @@
         {
           "id": "librarydataprovider",
           "name": "%views.SAS.libraries%",
-          "when": "(SAS.authorized && SAS.connectionType == rest) || SAS.connection.direct"
+          "when": "((SAS.authorized || SAS.librariesDisplayed) && SAS.librariesEnabled) || SAS.connection.direct"
         },
         {
           "id": "sas-content-get-started",
           "name": "%views.SAS.signIn%",
-          "when": "(!SAS.authorized && SAS.connectionType == rest) && !SAS.connection.direct"
+          "when": "(!(SAS.authorized || SAS.librariesDisplayed) && SAS.canSignIn) && !SAS.connection.direct"
         },
         {
           "id": "sas-content-invalid-connection",
           "name": "%views.SAS.signIn%",
-          "when": "!SAS.authorized && SAS.connectionType != rest"
+          "when": "!SAS.authorized && !SAS.canSignIn"
         }
       ]
     },
     "viewsWelcome": [
       {
         "view": "sas-content-get-started",
-        "when": "(!SAS.authorized && SAS.connectionType == rest) && !SAS.connection.direct",
         "contents": "%views.SAS.welcome%",
         "enablement": "!SAS.authorizing"
       },
       {
         "view": "sas-content-invalid-connection",
-        "when": "SAS.connectionType != rest",
```

Comment:
what about ssh?

Reply:
ssh would be an "invalid connection" (not the right wording) because it doesn't have access to sas content _or_ libraries/tables. Also, see the above :)